onnxruntime/core/providers/coreml/builders/impl/cast_op_builder.cc

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "core/providers/shared/utils/utils.h"
+#include "core/providers/coreml/builders/helper.h"
+#include "core/providers/coreml/builders/model_builder.h"
+#include "core/providers/coreml/builders/op_builder_factory.h"
+
+#include "base_op_builder.h"
+
+namespace onnxruntime {
+namespace coreml {
+
+class CastOpBuilder : public BaseOpBuilder {
+  // Add operator related
+ private:
+  Status AddToModelBuilderImpl(ModelBuilder& model_builder, const Node& node,
+                               const logging::Logger& logger) const override ORT_MUST_USE_RESULT;
+
+  // Operator support related
+ private:
+  bool IsOpSupportedImpl(const Node& node, const OpBuilderInputParams& input_params,
+                         const logging::Logger& logger) const override;
+};
+
+// Add operator related
+
+Status CastOpBuilder::AddToModelBuilderImpl(ModelBuilder& /* model_builder */,
+                                            const Node& /* node */,
+                                            const logging::Logger& /* logger */) const {
+  // Right now we're only handling an ArgMax op followed by a Cast to int32 type.
+  // This can fuse the ArgMax's int64 output type which is not supported in CoreML model.
+  // And that ArgMax fused with the cast node produces an int32 output, so we're skipping adding the Cast node here.
+  return Status::OK();
+}
+
+// Operator support related
+
+bool CastOpBuilder::IsOpSupportedImpl(const Node& node, const OpBuilderInputParams& input_params,
+                                      const logging::Logger& logger) const {
+  if (node.GetInputEdgesCount() == 0) {
+    LOGS(logger, VERBOSE) << "Failed to get Cast's preceding nodes.";
+    return false;
+  }
+
+  if (node.GetInputEdgesCount() > 1) {
+    LOGS(logger, VERBOSE) << "Multiple nodes consuming Cast's output.";
+    return false;
+  }
+
```

Comment:
Error message talks about output edges but 'if' condition is checking input edges.

Reply:
Sorry, I meant to check for the preceding nodes of Cast(meant to double check for the input edges for cast node). But the log message here is wrong. should be something like "multiple preceding nodes of Cast.."